File: android/app/src/main/java/com/jduplessis/results/api/Results.java
package com.jduplessis.results.api;

import com.google.api.client.json.GenericJson;
import com.google.api.client.util.DateTime;
import com.google.api.client.util.Key;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Janic on 2015-01-25.
 */
public class Results {
    @Key
    public DateTime lastUpdate;

    @Key
    public List<Class> classes;

    public static class Class extends GenericJson {
        @Key
        public String id;

        @Key
        public String name;

        @Key
        public String group;

        @Key
        public String year;

        @Key
        public List<Result> results;

        @Key
        public ResultInfo total;

        @Key("final")
        public String finalGrade;
    }

    public static class Result extends GenericJson {
        @Key
        public String name;

        @Key
        public ResultInfo normal;

        @Key
        public ResultInfo weighted;
    }

    public static class ResultInfo extends GenericJson {
        @Key
        public String result;

        @Key
        public String average;

        @Key
        public String standardDev;
    }
}

File: ios/iOsResultsCrawler/Model/Results.swift
//
//  Results.swift
//  iOsResultsCrawler
//
//  Created by Janic Duplessis on 2015-02-22.
//  Copyright (c) 2015 App and Flow. All rights reserved.
//

import Foundation

struct Results {
    var lastUpdate: String
    var classes: [Class]
}

struct Class {
    var id: String
    var name: String
    var group: String
    var results: [Result]
    var total: ResultInfo
    var finalGrade: String
}

struct Result {
    var name: String
    var normal: ResultInfo
    var weighted: ResultInfo
}

struct ResultInfo {
    var result: String
    var average: String
    var standardDev: String
}